### DISCUSSION TITLE
Opensumi可能不支持某些vscode扩展。

### DISCUSSION THREAD

**Discussion by 10000l** [opensumi/core#1885] [Q&A] [answered]
最近尝试安装python扩展，发现当前版本的opensumi安装了最新版本python扩展之后无法支持python的快速填充、自动引入、以及错误提示，官方除了降低插件版本是否还有其他方案，因为按照目前opensumi的扩展市场开发人员自行安装的话只能安装最新的，除非我们在extensions.json中设置好该插件，但是这样子不够灵活，有其他插件也可能有类似情况。

**Comment by erha19** [comment]
@linxiaolx-lx 可以尝试升级一下 OpenSumi 版本至 2.21 , 我们针对部分不支持的 notebook API 进行了空实现，除了 Notebook 这块功能，你应该可以使用 Python 插件的新功能。

**Reply by 10000l** [reply]
<img width="601" alt="image" src="https://user-images.githubusercontent.com/60864454/199191408-9382b434-feee-47d0-a648-1c01f0ef56e8.png">
更新至2.21版本之后会默认安装了python插件，在编辑python文件时会提示
`Extension activation failed, run the 'Developer: Toggle Developer Tools' command for more information.`
没有支持代码提示
<img width="377" alt="image" src="https://user-images.githubusercontent.com/60864454/199192281-da20eb92-7b59-48fe-8350-5b7d5ed7150f.png">
正常vscode在安装了扩展之后会有提示,如下：
<img width="548" alt="image" src="https://user-images.githubusercontent.com/60864454/199192766-e5f98817-f67c-4a2b-a594-d799012aa988.png">
<img width="303" alt="image" src="https://user-images.githubusercontent.com/60864454/199192858-f610e90e-e1d7-4435-b804-faf38d03dbad.png">

**Reply by 10000l** [reply]
我当前使用的是ide-startup最新版本更新了opensumi依赖到2.21
` 
    "@opensumi/ide-addons": "2.21.0",
    "@opensumi/ide-comments": "2.21.0",
    "@opensumi/ide-core-browser": "2.21.0",
    "@opensumi/ide-core-common": "2.21.0",
    "@opensumi/ide-core-node": "2.21.0",
    "@opensumi/ide-debug": "2.21.0",
    "@opensumi/ide-decoration": "2.21.0",
    "@opensumi/ide-editor": "2.21.0",
    "@opensumi/ide-explorer": "2.21.0",
    "@opensumi/ide-express-file-server": "2.21.0",
    "@opensumi/ide-extension": "2.21.0",
    "@opensumi/ide-extension-manager": "2.21.0",
    "@opensumi/ide-extension-storage": "2.21.0",
    "@opensumi/ide-file-scheme": "2.21.0",
    "@opensumi/ide-file-search": "2.21.0",
    "@opensumi/ide-file-service": "2.21.0",
    "@opensumi/ide-file-tree-next": "2.21.0",
    "@opensumi/ide-i18n": "2.21.0",
    "@opensumi/ide-keymaps": "2.21.0",
    "@opensumi/ide-logs": "2.21.0",
    "@opensumi/ide-main-layout": "2.21.0",
    "@opensumi/ide-markdown": "2.21.0",
    "@opensumi/ide-markers": "2.21.0",
    "@opensumi/ide-menu-bar": "2.21.0",
    "@opensumi/ide-monaco": "2.21.0",
    "@opensumi/ide-monaco-enhance": "2.21.0",
    "@opensumi/ide-opened-editor": "2.21.0",
    "@opensumi/ide-outline": "2.21.0",
    "@opensumi/ide-output": "2.21.0",
    "@opensumi/ide-overlay": "2.21.0",
    "@opensumi/ide-preferences": "2.21.0",
    "@opensumi/ide-process": "2.21.0",
    "@opensumi/ide-quick-open": "2.21.0",
    "@opensumi/ide-remote-opener": "2.21.0",
    "@opensumi/ide-scm": "2.21.0",
    "@opensumi/ide-search": "2.21.0",
    "@opensumi/ide-startup": "2.21.0",
    "@opensumi/ide-static-resource": "2.21.0",
    "@opensumi/ide-status-bar": "2.21.0",
    "@opensumi/ide-storage": "2.21.0",
    "@opensumi/ide-task": "2.21.0",
    "@opensumi/ide-terminal-next": "2.21.0",
    "@opensumi/ide-theme": "2.21.0",
    "@opensumi/ide-toolbar": "2.21.0",
    "@opensumi/ide-userstorage": "2.21.0",
    "@opensumi/ide-variable": "2.21.0",
    "@opensumi/ide-webview": "2.21.0",
    "@opensumi/ide-workspace": "2.21.0",
    "@opensumi/ide-workspace-edit": "2.21.0",
`

**Reply by erha19** [reply]
@linxiaolx-lx 插件不会默认安装，是你上个版本安装后遗留的，没有修改的话一般是在 `~/.sumi/extensions` 下，提供一下你这边使用的插件名称以及版本 ~ 我尝试复现一下

**Reply by 10000l** [reply]
> @linxiaolx-lx 插件不会默认安装，是你上个版本安装后遗留的，没有修改的话一般是在 `~/.sumi/extensions` 下，提供一下你这边使用的插件名称以及版本 ~ 我尝试复现一下

名称是:ms-python.python ,版本：2022.16.0。我下午重新clone了新项目，清理过之前安装的extensions的

**Comment by erha19** [comment, marked as answer]
@linxiaolx-lx 排查到原因了，主要原因是你安装的 python 插件版本对内核版本有要求，报错如下：
<img width="583" alt="image" src="https://user-images.githubusercontent.com/9823838/199201344-bf09d17e-85c9-4d55-9158-1d216523a41c.png">

目前 OpenSumi 的内核版本还在 1.63.2，这里的任务 https://github.com/opensumi/core/projects/17 完成后我们会将内核版本升级到 1.69.0

如果你目前需要使用该插件，有两种方式：

1. 修改插件内核版本
2. 修改集成侧的内核版本

修改集成侧的内核版本方法如下：

将原有插件进程的入口文件 `ext.process-base.js` 引出后重新定义内容后使用

```ts
import { extProcessInit } from '@opensumi/ide-extension/lib/hosted/ext.process-base.js';
import LogServiceClass from './mock-log-service';

(async () => {
  await extProcessInit({
    LogServiceClass,
    builtinCommands: [],
    customVSCodeEngineVersion: '1.67.0',
  });
})();
```

修改后插件运行效果如下：
<img width="1022" alt="image" src="https://user-images.githubusercontent.com/9823838/199203590-acb969ee-e47c-4453-865c-57a29d4c251b.png">

**Reply by 10000l** [reply]
收到，感谢
